syntax = "proto3";

package proto;

import "stalk_proto/google/api/annotations.proto";
import "stalk_proto/models.proto";

option go_package = "github.com/peake100/stalkforecaster-go/proto";


enum ImageFormat {
  SVG = 0;
  PNG = 1;
}


// Request message for creating a turnip forecast chart.
message ForecastChartReq {
  // Price ticker of the current prices on the island.
  Ticker ticker = 1;
  // The forecast for future prices on the the island.
  Forecast forecast = 2;
  // desired image format
  ImageFormat format = 3;
}


// Response message with a rendered svg chart.
message ChartResp {
  // Binary image bytes for rendered chart.
  bytes chart = 1;
}


// The reporting service definition.
service StalkReporter {
  // Takes a forecast and turns it into a pretty report
  rpc ForecastChart (ForecastChartReq) returns (ChartResp) {
    option (google.api.http) = {
      post: "/charts/forecast"
      body: "*"
    };
  }
}
